src/platform/nm-platform.c

```diff
@@ -3545,17 +3512,16 @@ gboolean
 nm_platform_ip6_address_sync (NMPlatform *self,
                               int ifindex,
                               GPtrArray *known_addresses,
-                              gboolean keep_link_local)
+                              gboolean full_sync)
 {
 	gs_unref_ptrarray GPtrArray *plat_addresses = NULL;
-	NMPlatformIP6Address *address;
 	gint32 now = nm_utils_get_monotonic_timestamp_s ();
-	int i, position;
+	guint i_plat, i_know;
+	gs_unref_hashtable GHashTable *known_addresses_idx = NULL;
 	NMPLookup lookup;
 	guint32 ifa_flags;
-	gboolean remove = FALSE;
 
-	if (!_addr_array_clean_expired (AF_INET6, ifindex, known_addresses, now, NULL))
+	if (!_addr_array_clean_expired (AF_INET6, ifindex, known_addresses, now, &known_addresses_idx))
 		known_addresses = NULL;
 
 	/* @plat_addresses and @known_addresses are in increasing priority order */
```

Comment:
The comment seems now obsolete, as the commit messages says:

> Note that @known_addresses contains the desired addresses in order of
    least-important first, while @plat_addresses contains them in opposite
    order.